### PR TITLE
OpenAMP Upgrade Part1: prepare for the OpenAMP Upgrade

### PR DESCRIPTION
**Pull request by CV-Bowen**
## Summary
1.    rpmsg_port.c:add rpmsg_port_dump function
    
    [01/01 00:02:48] [187] [ EMERG] [ap] Remote: wit
    [01/01 00:02:48] [187] [ EMERG] [ap] ept NS
    [01/01 00:02:48] [187] [ EMERG] [ap] ept rpmsg-sensor
    [01/01 00:02:48] [187] [ EMERG] [ap] ept rpmsg-ping
    [01/01 00:02:48] [187] [ EMERG] [ap] rpmsg_port queue RX: {used: 0, avail: 5}
    [01/01 00:02:48] [187] [ EMERG] [ap] rpmsg buffer list:
    [01/01 00:02:48] [187] [ EMERG] [ap] rpmsg_port queue TX: {used: 4, avail: 0}
    [01/01 00:02:48] [187] [ EMERG] [ap] rpmsg buffer list:
    [01/01 00:02:48] [187] [ EMERG] [ap]  TX buffer 0x619b84a8 hold by rpmsg-ping
    [01/01 00:02:48] [187] [ EMERG] [ap]  TX buffer 0x619b86a8 hold by rpmsg-ping
    [01/01 00:02:48] [187] [ EMERG] [ap]  TX buffer 0x619b88a8 hold by rpmsg-ping
    [01/01 00:02:48] [187] [ EMERG] [ap]  TX buffer 0x619b8aa8 hold by rpmsg-ping
    
2.    openamp/Kconfig: add CONFIG_OPENAMP_ENABLE_VIRTIO_ONLY to enable virtio only
    
    OPENAMP_VIRTIO_DEVICE_ONLY and OPENAMP_VIRTIO_DRIVER_ONLY can be used to
    comment some unused code in OpenAMP library to save code size.

3.    openamp: Refine elf_load to remoteproc_elf_loader
    
    to avoid the name conflict with binfmt elf loader
    
4.    openamp: Merge OPENAMP_RPMSG_DEBUG and OPENAMP_VQUEUE_DEBUG into OPENAMP_DEBUG
    
    to simplify the configuration

## Impact
openamp make and rpmsg port

## Testing
sim rpserver and sim rpproxy and bes chip

### CODE REVIEW

File: openamp/open-amp.defs [hunk 1]
```diff
@@ -26,12 +26,18 @@ ifeq ($(CONFIG_OPENAMP_CACHE),y)
...
-ifeq ($(CONFIG_OPENAMP_VQUEUE_DEBUG),y)
-  CFLAGS += -DVQUEUE_DEBUG
+CFLAGS += -Delf_load=remoteproc_elf_load
+
+ifeq ($(CONFIG_OPENAMP_VIRTIO_DEVICE_ONLY),y)
+  CFLAGS += -DVIRTIO_DEVICE_ONLY
+endif
+
+ifeq ($(CONFIG_OPENAMP_VIRTIO_DRIVER_ONLY),y)
+  CFLAGS += -DVIRTIO_DRIVER_ONLY
```
Comment (anchao) [inline openamp/open-amp.defs:40]: sync the makefile changes to CMake

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR summary you provided does meet some of the NuttX requirements, but it could be improved. Here's a breakdown:

**Strengths:**

* **Summary:**  The summary does a decent job of listing out the changes made. Each change is described briefly.
* **Testing:** You specify the testing environment and mention having logs (although not included here).

**Areas for Improvement:**

* **Summary:**
    * **Why:** The summary lacks a clear explanation of *why* these changes are necessary. Are they bug fixes? Performance improvements? New features? Be explicit about the motivation behind each change.
    * **How:** While you mention what changes, explaining *how* those changes work would be beneficial. For example, for point 1, what does `rpmsg_port_dump` output, and how does it help debug? 
* **Impact:**
    * **Too brief:** The "Impact" section is too general. You need to address each bullet point individually. 
    * **User Impact:**  Will users need to change their code or configuration with this change? If so, how?
    * **Build Impact:**  Do any build dependencies change? Will users need to modify their build system?
    * **Hardware Impact:**  Be specific about which architectures, boards, and drivers are affected (if any).
    * **Documentation:** If documentation needs to be updated, mention what needs to be changed.  If you've included documentation updates in the PR, state that explicitly.
    * **Security, Compatibility:** Even if there is no impact on these, explicitly stating "NO" for each is helpful for reviewers.
* **Testing:**
    * **Include Logs (if possible):**  While you mention having logs, including snippets of the logs (before and after) would greatly help reviewers understand the changes and their impact.
    * **Test Cases:** Briefly describe the specific test cases you used to verify the changes.

**Revised Summary Example (for point 1):**

1. **Add `rpmsg_port_dump` function to `rpmsg_port.c`:** This change adds a new function to aid in debugging Remote Processor Messaging (RPMSG) communication issues.  The `rpmsg_port_dump` function prints the current state of the RPMSG port, including the number of buffers used and available in the RX and TX queues, as well as information about any buffers currently held by specific endpoints. This information can be invaluable when troubleshooting RPMSG communication problems.

**By addressing these points and providing more specific information, you can make your NuttX PR much stronger and easier for reviewers to understand and approve.**

**Review by jerpelea — CHANGES_REQUESTED** [review]
please add a commit message with explanation for this change

**Comment by xiaoxiang781216** [comment]
> please add a commit message with explanation for this change

ALL change is minor, and the commit message already explain the change. So, could you point out which patch need provide more info to help you review the change?

**Comment by jerpelea** [comment]
@xiaoxiang781216 the explanation is added to the PR message not the individual commits. after merge the PR message is not visible with git log and it will be  fairly hard to understand the reason for chains in 2-3 years, without context.

**Comment by jerpelea** [comment]
@xiaoxiang781216 if you check the release notes for 12.7.0 you will understand why I am pushing for a bit better commit documentation

**Review by jerpelea — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**